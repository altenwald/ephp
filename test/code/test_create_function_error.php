<?php
$newfunc = create_function('$a,$b', 'return "ln($a) + ln($b) = " . log($a * $b);');
echo "New anonymous function: $newfunc\n";
echo $newfunc(2, M_E) . "\n";

$closure = function($a, $b) { return "ln($a) + ln($b) = " . log($a * $b); };
echo "New closure: $closure\n";
echo $closure(2, M_E) . "\n";

var_dump($newfunc);
var_dump($closure);
