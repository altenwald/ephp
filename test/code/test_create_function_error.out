New anonymous function: lambda_1
ln(2) + ln(2.71828182845905) = 1.69314718055995

Catchable fatal error: Object of class Closure could not be converted to string in {{CWD}}/test/code/test_create_function_error.php on line 7
